### PR TITLE
WIC and ECE updates

### PR DESCRIPTION
**Pull request by spokenbird**

### CODE REVIEW

File: src/main/java/org/ladocuploader/app/csv/model/StudentCsvModel.java [hunk 1]
```diff
@@ -102,13 +106,18 @@ public static List<BaseCsvModel> generateModel(Submission submission){
 
         if (household != null && !household.isEmpty()) {
             for (Map<String, Object> member : household) {
-                String relationship = (String) member.get("householdMemberRelationship");
-                if (relationship == null || !relationship.equalsIgnoreCase("child")) {
+                //String relationship = (String) member.get("householdMemberRelationship");
```
Comment (bseeger) [inline src/main/java/org/ladocuploader/app/csv/model/StudentCsvModel.java:109]: I think you can take this commented out code out -- it was replaced by the "isMemberEceEligible" code.

Comment (bseeger) [reply src/main/java/org/ladocuploader/app/csv/model/StudentCsvModel.java:109]: If we are mapping this object, the applicant was interested in ECE, so the "isMemberEceEligible" will check to see if the household member is either pregnant or young enough to be the student.  If so, it will add them as a student.

File: src/main/java/org/ladocuploader/app/FileExportController.java [hunk 2]
```diff
@@ -70,13 +71,12 @@ ResponseEntity<?> downloadPGCsv(
         Locale locale
     ) throws IOException, CsvRequiredFieldEmptyException, CsvDataTypeMismatchException {
 
-        String cleanedFlow = StringEscapeUtils.escapeHtml4(flow);
-        String cleanedSubmissionId = StringEscapeUtils.escapeHtml4(submissionId);
+        String encodedFlow = UriComponentsBuilder.fromPath(flow).build().toUriString();
```
Comment (sree-cfa) [inline src/main/java/org/ladocuploader/app/FileExportController.java:74]: What does "encoded" mean here?

File: src/main/java/org/ladocuploader/app/csv/converters/UnmarriedMinorConverter.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import org.ladocuploader.app.utils.HouseholdUtilities;
+import java.time.LocalDate;
+import java.util.Map;
+
+/**
+ * This converter will take marital status and age and compute unmarried minor
+ */
+@NoArgsConstructor
+@Slf4j
+public class UnmarriedMinorConverter<T, I> extends AbstractBeanField<T, I> {
```
Comment (sree-cfa) [inline src/main/java/org/ladocuploader/app/csv/converters/UnmarriedMinorConverter.java:15]: uh, what

File: src/main/java/org/ladocuploader/app/FileExportController.java [hunk 4]
```diff
@@ -149,8 +182,8 @@ private ResponseEntity handleCsvGeneration(String flow, String submissionId, Htt
                     .headers(headers)
                     .body(csvDoc.getCsvData());
         } else {
-            log.error("Attempted to download PDF with submission_id: " + submissionId + " but session_id was: "
-                    + httpSession.getAttribute("id"));
+            log.error("Attempted to download PDF with submission_id: " + submissionIdSanitized + " but session_id was: "
+                    + sanitizeSubmissionId(httpSession.getAttribute("id").toString()));
```
Comment (sree-cfa) [inline src/main/java/org/ladocuploader/app/FileExportController.java:186]: What are we checking here with `httpSession.getAttribute("id")`? In the DataInterceptor, it looks like theres a flow-to-submissionID map that we keep the submission attributes in - is that related to this?

File: src/main/java/org/ladocuploader/app/utils/HouseholdUtilities.java [hunk 5]
```diff
@@ -1,34 +1,82 @@
...
 
     if (day <= 0 || month <= 0 || year <= 0 ) {
       throw new NumberFormatException("cannot analyze birthdate as fields are missing");
     }
 
-    Calendar memberBirthDayCal = Calendar.getInstance();
-    memberBirthDayCal.set(Calendar.YEAR, year);
-    memberBirthDayCal.set(Calendar.MONTH, month);
-    memberBirthDayCal.set(Calendar.DAY_OF_MONTH, day);
+    Calendar memberBirthDayCal = new Calendar.Builder().setDate(year, month, day).build();
```
Comment (sree-cfa) [inline src/main/java/org/ladocuploader/app/utils/HouseholdUtilities.java:39]: `LocalDate` has "isBefore" and "isAfter" methods that might be more readable here.

File: src/main/java/org/ladocuploader/app/utils/HouseholdUtilities.java [hunk 6]
```diff
@@ -1,34 +1,82 @@
...
+   * @throws NumberFormatException throws this if the day, month or year passed in are invalid
+   */
+  public static boolean isMemberEceEligible(Map<String, Object> member, Map<String, Object> inputData) throws NumberFormatException {
+    boolean isPregnant = ((List)inputData.getOrDefault("pregancies[]", List.of())).contains(member.get("uuid"));
+
+    if (isPregnant) {
+      return true;
+    }
+
+    int birthDay = Integer.parseInt((String) member.get("householdMemberBirthDay"));
```
Comment (sree-cfa) [inline src/main/java/org/ladocuploader/app/utils/HouseholdUtilities.java:65]: Does `parseInt` work if DOB is not provided? I can't remember if this is a required field or not

File: src/main/java/org/ladocuploader/app/FileExportController.java (hunk 2 above)
Comment (spokenbird) [reply src/main/java/org/ladocuploader/app/FileExportController.java:74]: Uri Encoded, where the characters are escaped based on URL params. 

- A space character (" ") is replaced by "%20".
- An ampersand ("&") is replaced by "%26".

File: src/main/java/org/ladocuploader/app/csv/converters/UnmarriedMinorConverter.java (hunk 3 above)
Comment (spokenbird) [reply src/main/java/org/ladocuploader/app/csv/converters/UnmarriedMinorConverter.java:15]: @lkemperman-cfa ? 😅

Comment (lkemperman-cfa) [reply src/main/java/org/ladocuploader/app/csv/converters/UnmarriedMinorConverter.java:15]: lol should this be renamed? this is literally the question on the ECE app :/

File: src/main/java/org/ladocuploader/app/utils/HouseholdUtilities.java (hunk 5 above)
Comment (spokenbird) [reply src/main/java/org/ladocuploader/app/utils/HouseholdUtilities.java]: Pushed up a change based on this feedback

File: src/main/java/org/ladocuploader/app/utils/HouseholdUtilities.java (hunk 6 above)
Comment (spokenbird) [reply src/main/java/org/ladocuploader/app/utils/HouseholdUtilities.java:62]: Will ask Roshan about this in the morning, and what should happen if it's not provided

Comment (spokenbird) [reply src/main/java/org/ladocuploader/app/utils/HouseholdUtilities.java:62]: Ignoring this as we are making these fields required as part of another ticket

File: src/main/java/org/ladocuploader/app/FileExportController.java (hunk 4 above)
Comment (spokenbird) [reply src/main/java/org/ladocuploader/app/FileExportController.java:186]: @bseeger do you have the context here?

Comment (bseeger) [reply src/main/java/org/ladocuploader/app/FileExportController.java:186]: Looks like it's trying to grab the submission id the old way, where as it needs to grab it from a flow-to-submissionId map as @sree-cfa suggests.    Not really sure what Lauren's goal was here, but since the `if` was for ensuring the submission id == the one in the session, I'm guessing she meant to pull the id from the map that Sree mentioned.

Comment (spokenbird) [reply src/main/java/org/ladocuploader/app/FileExportController.java:186]: Sorry I should have tagged @lkemperman-cfa here 😅

Comment (lkemperman-cfa) [reply src/main/java/org/ladocuploader/app/FileExportController.java:186]: Hi! That is not supposed to be there (should be fetching submission id the new way), but the sanitize method is to avoid log injection security issue.
